### PR TITLE
Configure sub-store mapping for models

### PR DESCRIPTION
**Pull request by slimjim777**
The sub-store mapping allows a model to be pivoted to a sub-store with a new model name

### CODE REVIEW

File: service/handlers_substore.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017-2018 Canonical Ltd
```
Comment (rmescandon) [inline service/handlers_substore.go]: I wonder if setting only 2018 for new files would be better

File: service/handlers_substore_test.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+var _ = check.Suite(&SubstoreSuite{})
+
+func (s *SubstoreSuite) SetUpTest(c *check.C) {
+	// Mock the database
+	config := config.Settings{KeyStoreType: "filesystem", KeyStorePath: "../keystore", JwtSecret: "SomeTestSecretValue"}
+	datastore.Environ = &datastore.Env{DB: &datastore.MockDB{}, Config: config}
+	datastore.OpenKeyStore(config)
+}
+
+func (s *SubstoreSuite) sendRequest(method, url string, data io.Reader, permissions int, c *check.C) *httptest.ResponseRecorder {
```
Comment (rmescandon) [inline service/handlers_substore_test.go]: Actually you don't need sendRequest, parseSubstoresResponse or parseBooleanResponse to have a pointer receiver to the suite, they are not using anything from the Suite

File: webapp-admin/src/components/AccountDetail.js [hunk 3]
```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (C) 2017-2018 Canonical Ltd
```
Comment (rmescandon) [inline webapp-admin/src/components/AccountDetail.js]: 2018?

File: webapp-admin/src/components/SubstoreForm.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2017-2018 Canonical Ltd
```
Comment (rmescandon) [inline webapp-admin/src/components/SubstoreForm.js]: 2018

File: service/handlers_substore.go (hunk 1 above)
Comment (slimjim777) [reply service/handlers_substore.go]: Okay, I can do that

**Pull request merged**